lib/srv/server/ec2_watcher.go

```diff
@@ -52,7 +53,69 @@ type EC2Instances struct {
 	// AccountID is the AWS account the instances belong to.
 	AccountID string
 	// Instances is a list of discovered EC2 instances
-	Instances []*ec2.Instance
+	Instances []EC2Instance
+}
+
+type EC2Instance struct {
+	InstanceID string
+}
+
+func toEC2Instance(inst *ec2.Instance) EC2Instance {
+	return EC2Instance{
+		InstanceID: aws.StringValue(inst.InstanceId),
+	}
+}
+
+// ToEC2Instances converts aws []*ec2.Instance to []EC2Instance
+func ToEC2Instances(insts []*ec2.Instance) []EC2Instance {
+	var ec2Insts []EC2Instance
+
+	for _, inst := range insts {
+		ec2Insts = append(ec2Insts, toEC2Instance(inst))
+	}
+	return ec2Insts
+
+}
+
+func NewEC2AgentlessWatcher(ctx context.Context, rotationEvent chan struct{}, matchers []services.AWSMatcher, clients cloud.Clients, nodeWatcher *services.NodeWatcher, ap auth.DiscoveryAccessPoint) (*Watcher, error) {
```

Comment:
Still don't understand why do we need a whole separate watcher while we already have it basically implemented for EC2 discovery.